src/e2e/app/connection-dialogs.ts

```diff
@@ -0,0 +1,151 @@
+/**
+ * (c) 2024, Micro:bit Educational Foundation and contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+import { expect, Locator, type Page } from "@playwright/test";
+import { MockWebUSBConnection } from "../../device/mockUsb";
+import { MockWebBluetoothConnection } from "../../device/mockBluetooth";
+import { ConnectionStatus } from "@microbit/microbit-connection";
+
+interface DialogStep {
+  text: string;
+  next?: "copy-pattern";
+}
+
+const dialogTypes: {
```

Comment:
I am unsure that the abstraction and looping here is a good idea...

Is there a simpler version of this that's more straight line test code? Does it have unacceptable duplication?